tests/js/client/shell/shell-index.js

```diff
@@ -0,0 +1,116 @@
+/*jshint globalstrict:false, strict:false */
+/*global fail, assertEqual, assertNotEqual, assertTrue, assertFalse, assertNull */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test index methods
+///
+/// DISCLAIMER
+///
+/// Copyright 2010-2012 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Dr. Frank Celler, Lucas Dohmen
+/// @author Copyright 2012, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const arango = internal.arango;
+const errors = internal.errors;
+const print = internal.print;
+const { helper, versionHas } = require("@arangodb/test-helper");
+const platform = require('internal').platform;
+
+
+const cn = "UnitTestsCollection";
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite: basics
+////////////////////////////////////////////////////////////////////////////////
+
+function IndexSuite() {
+  'use strict';
+
+  let collection = null;
+
+  return {
+
+    setUp : function () {
+      internal.db._drop(cn);
+      collection = internal.db._create(cn);
+    },
+
+    tearDown : function () {
+      internal.db._drop(cn);
+      collection = null;
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test: indexes
+////////////////////////////////////////////////////////////////////////////////
+
+    testIndexProgress : function () {
+      var c = internal.db._create("c0l", {numberOfShards: 3, replicationFactor: 2});
+      var docs = [];
+      const sleep = require('internal').sleep;
+      for (var j = 0; j < 16; ++j) {
+        docs=[];
+        for (var i = 0; i < 10240; ++i) {
+          docs.push({name : "name" + i});
+        }
+        c.insert(docs);
+      }
+      
+      internal.debugSetFailAt("fillIndex::pause");
```

Comment:
This will work in single server, because `debugSetFailAt` will set the failure point on the point of contact, which is the coordinator. For the cluster, it is more difficult: We have to first find the endpoints of all dbservers and then use the two argument version in `test-helpers` to activate the failure points there. And likewise in the end switch off the failure points again. I bet that on cluster this only works since the test can never fail.